android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java

```diff
@@ -0,0 +1,329 @@
+        alarmMgr = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+        List<DSTData> dstList = checkTimeZoneIsOnDSTAndGetDSTData(context, TimeZone.getDefault().getID());
+        Calendar currentCalendar = Calendar.getInstance();
+        int dst_start_request_code = DST.DST_START.getValue();
+        int dst_end_request_code = DST.DST_END.getValue();
+        /* we are using loops because if we have countries with changing dst-start/end date,
+        ie; iran have 21/22 march as dst start date*/
+        for (DSTData dstObject : dstList) {
+            int startMonth = dstObject.getStart_month_index();
+            int endMonth = dstObject.getEnd_month_index();
+            int startTime = dstObject.getStart_time();
+            int endTime = dstObject.getEnd_time();
+            int year = currentCalendar.get(Calendar.YEAR);
+
+            int startDay;
+            int endDay;
+            if (dstObject.getStart_week_day() != null & dstObject.getEnd_week_day() != null) {
+                String startWeekDay = dstObject.getStart_week_day().toUpperCase();
+                String endWeekDay = dstObject.getEnd_week_day().toUpperCase();
+                startDay = getDayFromWeekDay(startWeekDay, startMonth, year);// ie; LSAT_SUNDAY 28
+                endDay = getDayFromWeekDay(endWeekDay, startMonth, year);// ie; LAST_SUNDAY 28
+            } else {
+                startDay = dstObject.getStart_day();
+                endDay = dstObject.getEnd_day();
+            }
+
+            Calendar startCalendar = Calendar.getInstance();
+            startCalendar.set(Calendar.MONTH, startMonth);
+            startCalendar.set(Calendar.DAY_OF_MONTH, startDay);
+            startCalendar.set(Calendar.HOUR_OF_DAY, startTime);
+            startCalendar.set(Calendar.MINUTE, 0);
+            startCalendar.set(Calendar.SECOND, 0);
+            startCalendar.set(Calendar.MILLISECOND, 0);
+
+            Calendar endCalendar = Calendar.getInstance();
+            endCalendar.set(Calendar.MONTH, endMonth);
+            endCalendar.set(Calendar.DAY_OF_MONTH, endDay);
+            endCalendar.set(Calendar.HOUR_OF_DAY, endTime);
+            endCalendar.set(Calendar.MINUTE, 0);
+            endCalendar.set(Calendar.SECOND, 0);
+            endCalendar.set(Calendar.MILLISECOND, 0);
+
+            if (currentCalendar.before(startCalendar)) {
+                scheduleDSTBroadCast(context, dst_start_request_code, startCalendar);
+                dst_start_request_code++;// scheduling multiple notifications if countries
+            } else if (currentCalendar.before(endCalendar)) {
+                scheduleDSTBroadCast(context, dst_end_request_code, endCalendar);
+                dst_end_request_code++;
+            } else {
+                scheduleDSTBroadcastForNextYear(context);
+            }
+            resetNotificationDateInLibrary(context);
+        }
+    }
+
+    private void resetNotificationDateInLibrary(Context context) {
```

Comment:
Can we change the function name? This is already a library.

Reply:
Sorry, I didn't understand that. did you mean the function is in the library so we need to change that right?  if yes what should be the name? can we change it to resetNotificationDate?

Reply:
resetNotificationDate should be fine!